task-sdk/src/airflow/sdk/definitions/mappedoperator.py

```diff
@@ -144,6 +154,21 @@ def is_mappable_value(value: Any) -> TypeGuard[Collection]:
     return True
 
 
+def prevent_duplicates(kwargs1: dict[str, Any], kwargs2: Mapping[str, Any], *, fail_reason: str) -> None:
+    """
+    Ensure *kwargs1* and *kwargs2* do not contain common keys.
+
+    :raises TypeError: If common keys are found.
+    """
+    duplicated_keys = set(kwargs1).intersection(kwargs2)
+    if not duplicated_keys:
+        return
+    if len(duplicated_keys) == 1:
+        raise TypeError(f"{fail_reason} argument: {duplicated_keys.pop()}")
+    duplicated_keys_display = ", ".join(sorted(duplicated_keys))
+    raise TypeError(f"{fail_reason} arguments: {duplicated_keys_display}")
```

Comment:
Ditto here -- ValueError? (could be separate PR)

I also notice the original method from utils is still there -- is there anything else left using that version?

Reply:
Checked again, there was one occurence in core which I have already handled using a private method, so getting rid of that

Reply:
Deleted it and handled compat too. It works fine as shown in PR desc.